libraries/lib-mixer/MixerSource.h

```diff
@@ -112,6 +113,8 @@ class MixerSource final : public AudioGraph::Source {
 
    //! SampleTrackCache is the source of data
    std::unique_ptr<SampleTrackCache> mInputSequence;
+   std::vector<LibSampleTrack::TrackBlockSequence> mCacheHolders;
+   double mTimePos = 0.;
```

Comment:
This solution no longer intrudes in SampleTrack but still does intrude in lib-mixer.

I am not yet decided whether I like it.